src/Generator/SequentialInvoiceNumberGenerator.php

```diff
@@ -49,15 +49,22 @@ public function __construct(
 
     public function generate(): string
     {
-        $invoiceIdentifierPrefix = $this->dateTimeProvider->__invoke()->format('Y/m') . '/';
+        $currentDate = $this->dateTimeProvider->__invoke();
```

Comment:
This can also cause some issues, as you probably want the invoice number to be generated just when the order is placed / order payment is paid, so there's a rare edge case when the month changes between those. What do you think about modifying `InvoiceNumberGeneratorInterface` to accept `\DateTimeInterface` as an argument in `generate()` method, so that we won't need to rely on the current server time?